apps/calendar/js/utils/animated_scroll.js

```diff
@@ -0,0 +1,34 @@
+define(function(require, exports, module) {
+'use strict';
+
+module.exports = function animatedScroll(opts) {
+  var {container, content, scrollTop} = opts;
+
+  var maxScroll = container.scrollHeight - container.clientHeight;
+  scrollTop = Math.min(Math.max(scrollTop, 0), maxScroll);
```

Comment:
just in case the `scrollTop` is negative or `null` - I really miss a `Math.clamp(val, min, max)` function.